### PR TITLE
Add searchable feature for plugins on plublic repo

### PR DESCRIPTION
**Pull request by kibablu**
Add a searchable feature for plugins on public repo #439 which closes the earlier pr @jennydaman

### CODE REVIEW

File: chris_backend/plugins/models.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
-
+from cgitb import lookup
+from dataclasses import field
```
Comment (jennydaman) [inline chris_backend/plugins/models.py:2]: What are these two imports for?

File: chris_backend/plugins/models.py [hunk 2]
```diff
@@ -240,11 +242,19 @@ def search_name_title_category(self, queryset, name, value):
         lookup = lookup | models.Q(meta__category__icontains=value)
         return queryset.filter(lookup)
 
+
+   def search_name_public_rep(self, queryset, name, value):
```
Comment (jennydaman) [inline chris_backend/plugins/models.py:246]: What is the purpose of this function definition?

**Review by jennydaman — CHANGES_REQUESTED** [review]

File: chris_backend/plugins/models.py (hunk 1 above)
Comment (kibablu) [reply chris_backend/plugins/models.py:2]: Must have been picked by the `VsCode` extension ... I do not think they add on what I was trying to do. I have disabled the extension

File: chris_backend/plugins/models.py (hunk 2 above)
Comment (kibablu) [reply chris_backend/plugins/models.py:246]: Trying to return any value that contains `public_repo` when you search

File: chris_backend/plugins/models.py (hunk 1 above)
Comment (jennydaman) [reply chris_backend/plugins/models.py:2]: Remove these lines. I recommend you review your `git diff` before making commits.

File: chris_backend/plugins/models.py (hunk 2 above)
Comment (jennydaman) [reply chris_backend/plugins/models.py:246]: Where is being called?

File: chris_backend/plugins/models.py (hunk 1 above)
Comment (kibablu) [reply chris_backend/plugins/models.py]: Thank you, I have made chnages

File: chris_backend/plugins/models.py (hunk 2 above)
Comment (kibablu) [reply chris_backend/plugins/models.py]: My understanding is from `pluginMeta table`

**Review by jennydaman — APPROVED** [review]

**Review by jbernal0019 — APPROVED** [review]

**Review by jbernal0019 — CHANGES_REQUESTED** [review]
public_repo = django_filters.CharFilter(field_name='meta__type', lookup_expr='icontains')

should be changed to 

 public_repo = django_filters.CharFilter(field_name='meta__public_repo, lookup_expr='icontains')

**Comment by kibablu** [comment]
> public_repo = django_filters.CharFilter(field_name='meta__type', lookup_expr='icontains')
> 
> should be changed to
> 
> public_repo = django_filters.CharFilter(field_name='meta__public_repo, lookup_expr='icontains')

@jbernal0019 Thank you, I have made changes

**Pull request closed without merging**